### ISSUE TITLE
Re-introduce list search for blocked and blocked by

### ISSUE THREAD

**Issue by thieflord06** [ClearskyApp06/ClearskyUI#211]
We previously had to remove this feature but now that we are back stable we can bring it back. before, the search was client side because it was fetching the whole list, now it is dynamically fetching based on scroll so we will have to use the API endpoint to get the data. I would recommend we use the same autocomplete that is being used on the home page to get the handle, searching for partials is expensive.

Endpoint: /api/v1/anon/blocklist-search-blocked/<handle1>/<handle2>
Description: Get if handle 1 blocked by handle 2
Parameters: handle 1 & handle 2
Response:
    {
        "data":
            {
                "blocked_date":"2023-10-23T22:17:42.214000+00:00",
                "handle":"desir.ee",
                "status":true
            }
    }

/api/v1/anon/blocklist-search-blocking/<handle1>/<handle2>
Description: Get if handle 1 is blocked by handle 2
Parameters: handle 1 & handle 2

Response:
    {
        "data":
              {
                  "blocked_date":"2024-03-05T22:40:37.498000+00:00",
                  "handle":"mary.my.id",
                  "status":true
              }
    }

**Comment by thieflord06** [comment]
The endpoints will be changing to handle DIDs only (in progress)

**Comment by thieflord06** [comment]
![image](https://github.com/user-attachments/assets/5427e4e2-9873-4c79-ba4e-763e9ffa0235)

**Comment by rumanstheddy** [comment]
@thieflord06 can I work on this?

**Comment by thieflord06** [comment]
@rumanstheddy I just assigned to you. Please note the updated endpoint names. I just enabled the endpoint. 

Also, please add the feature flag framework for this, this is the name: ```blocking-searching```

**Comment by rumanstheddy** [comment]
Sure. Will do. Thanks!

**Comment by rumanstheddy** [comment]
hey @thieflord06 I integrated the autocomplete functionality to the blocking tabs.
Here's how it looks. Let me know if you want something changed.

https://github.com/user-attachments/assets/85abf05c-c1f5-4332-8ce2-a652fd797c37

**Comment by thieflord06** [comment]
This is taking them to the profile page if true is being returned from the endpoint for that search?

And if it is false then nothing shows in the search?

**Comment by rumanstheddy** [comment]
No, this uses a flatMap of the blocklist users to select the options. The flatMap keeps updating on scroll (for every new fetch). I tried implementing the approach which searches handles and filters them according to the blocking/blocked-by endpoints, it works but is extremely slow and leads to a 429 error.
Is there a particular reason why we need to make use of the endpoints for every search here? Would using the flatMap approach be acceptable for performance and UX reasons? Would like to know your thoughts.

EDIT: Nevermind. I re-implemented the approach that works with the `/blocklist-search-blocked` and `/blocklist-search-blocking` endpoints to filter the results returned from the search. I added concurrency workers and cache to fix the 429 errors.

**Comment by rumanstheddy** [comment]
There seems to be an issue with the `/blocklist-search-blocking` endpoint. For the account 'georgetakei', I noticed that the data being returned from the API is null, even though the handle I verified this against is in the blocking list.

Here's the example:
<img width="1759" height="524" alt="Image" src="https://github.com/user-attachments/assets/eaba608e-8c56-4828-b9fd-4439ce644735" />

Here's a working example:
<img width="1905" height="800" alt="Image" src="https://github.com/user-attachments/assets/4d01b78b-3bb2-4934-b971-092088438fa8" />

**Comment by rumanstheddy** [comment]
Hey @thieflord06 
Just following up on my previous post regarding the issue with the endpoint.
Could you please verify this and confirm?

**Comment by thieflord06** [comment]
@rumanstheddy I am taking a look at this now.

**Comment by thieflord06** [comment]
@rumanstheddy Can you make a PR so I can test this?

**Comment by rumanstheddy** [comment]
@thieflord06 I haven't cleaned up the code yet for a PR, but I've pushed my changes to my forked branch for you to take a look.
Also included the console.log statements, as shown [here](https://github.com/ClearskyApp06/ClearskyUI/issues/211#issuecomment-3576820537) in the file `blocking/index`, to help you test.
Here's the [link ](https://github.com/rumanstheddy/ClearskyUI/tree/feature/list-search-211) to my repo.
Please use the latest version from the branch `list-search-211`

**Comment by rumanstheddy** [comment]
Hey @thieflord06 I just cleaned up the code, pushed my changes, and opened a [PR](https://github.com/ClearskyApp06/ClearskyUI/pull/435).

**Comment by thieflord06** [comment]
@rumanstheddy Thank you. I'll troubleshoot and see what the fix should be.

**Comment by rumanstheddy** [comment]
Hi @thieflord06  Just checking in regarding this issue. 
I’m happy to pick it up and continue working on it whenever you’re ready. 
Please let me know if there’s anything else I can help with.

**Comment by thieflord06** [comment]
@rumanstheddy I just checked out this issue. It is working as expected. The staging is using VERY old data so that's why what you're searching is coming back with null. 

I enabled the endpoints for production so you can verify on there now.

**Comment by thieflord06** [comment]
@rumanstheddy Looks like there is a CORS issue. It's not happening locally and not happening in staging.
